src/peer_manager.rs

```diff
@@ -1217,11 +1219,14 @@ impl PeerManager {
 
         let _ = self.insert_peer(pub_id, Some(peer_id), PeerState::SearchingForTunnel);
 
-        let close_section = self.routing_table.other_close_names(pub_id.name()).unwrap_or_default();
+        let close_nodes = self.routing_table
```

Comment:
not quite follow that, but do you mean we switch to : 
let close_nodes = self.routing_table.other_close_names(self.public_id.name()) .. ;
then `.take(NUM_TUNNEL_VIA_NODES)` after peer_map check ?
i.e. we will use our_section or part of it if section_size > NUM_TUNNEL_VIA_NODES ?